nilearn/glm/tests/test_thresholding.py

```diff
@@ -437,6 +438,21 @@ def test_all_resolution_inference_threshold_errors(
         cluster_level_inference(stat_img, threshold=threshold)
 
 
+def test_all_resolution_inference_shape_errors(img_4d_rand_eye, surf_img_2d):
+    """Raise error with 4D image or SurfaceImage with n_samples>1."""
+    with pytest.raises(
+        DimensionError,
+        match=("Input data has incompatible dimensionality"),
+    ):
+        cluster_level_inference(img_4d_rand_eye, threshold=0.5)
```

Comment:
@bthirion we agree that cluster_level_inference should run on 3D images?
Note that it does fail currently on 4D images but the error is not very clear.

Reply:
Agreed.